### ISSUE TITLE
the relation attribute's entry title can't use id field in Content Manager

### ISSUE THREAD

**Issue by wusuopu** [strapi/strapi#17503]
### Required System information

- Node.js version: v16.13.2
- NPM version: 1.22.15
- Strapi version: 4.11.7
- Database: sqlite 3
- Operating system: MacOS 13.3
- Is your project Javascript or Typescript: Typescript

### Steps to reproduce the behavior

1.Add two content-types in new project. `Blog` has oneToOne relation to `Info`.
 ```json
{
  "kind": "collectionType",
  "collectionName": "blogs",
  "info": {
    "singularName": "blog",
    "pluralName": "blogs",
    "displayName": "Blog"
  },
  "options": {
    "draftAndPublish": false,
    "comment": ""
  },
  "attributes": {
    "info": {
      "type": "relation",
      "relation": "oneToOne",
      "target": "plugin::cms.info",
      "inversedBy": "blog",
      "useJoinTable": false
    },
    "title": {
      "type": "string"
    }
  }
}
```

```json
{
  "kind": "collectionType",
  "collectionName": "infos",
  "info": {
    "singularName": "info",
    "pluralName": "infos",
    "displayName": "Info"
  },
  "options": {
    "draftAndPublish": false,
    "comment": ""
  },
  "attributes": {
    "blog": {
      "type": "relation",
      "relation": "oneToOne",
      "target": "plugin::cms.blog",
      "mappedBy": "info"
    },
    "tag": {
      "type": "string"
    },
    "media": {
      "type": "text"
    }
  }
}
```

2.Configure the view for `Blog`, and change entry title of `info` attribute to `id` field.
![image](https://github.com/strapi/strapi/assets/1478284/fcd38c62-3069-4159-8eb8-8110e61a50b3)

3.Then edit a Blog record, and select any one `info` record.
![image](https://github.com/strapi/strapi/assets/1478284/4dbfbf19-78d5-4fef-9fc0-284542a2801a)

4.After select one `info` record, it will occur a error.
![image](https://github.com/strapi/strapi/assets/1478284/956f8727-e113-43a9-bae4-08870bc862dd)
the reason maybe is that current entry title is `id` field, and the value selected is a number not string. so that it call `normalize` function fail.

5.Then change entry title of `info` attribute back to `tag` field, it will occur other error.
![image](https://github.com/strapi/strapi/assets/1478284/61940344-e54e-4add-8898-069bd8c309bb)

6.Edit a Blog record again. it will occur a error when save.
![image](https://github.com/strapi/strapi/assets/1478284/bbcba02f-8e84-4721-99a6-e1e9b36bccfd)

the exception info in backend:
```
[2023-07-30 10:54:23.230] error: insert into `blogs` (`created_at`, `created_by_id`, `info_id`, `title`, `updated_at`, `updated_by_id`) values ('2023-07-30 10:54:23.229', 1, '{"disconnect":[],"connect":[{"id":1,"position":{"end":true}}]}', 'Blog Title1', '2023-07-30 10:54:23.229', 1) returning `id` - FOREIGN KEY constraint failed
SqliteError: insert into `blogs` (`created_at`, `created_by_id`, `info_id`, `title`, `updated_at`, `updated_by_id`) values ('2023-07-30 10:54:23.229', 1, '{"disconnect":[],"connect":[{"id":1,"position":{"end":true}}]}', 'Blog Title1', '2023-07-30 10:54:23.229', 1) returning `id` - FOREIGN KEY constraint failed
    at Client_BetterSQLite3._query (/Users/user1/strapi4-demo/node_modules/knex/lib/dialects/better-sqlite3/index.js:39:40)
    at executeQuery (/Users/user1/strapi4-demo/node_modules/knex/lib/execution/internal/query-executioner.js:37:17)
    at Client_BetterSQLite3.query (/Users/user1/strapi4-demo/node_modules/knex/lib/client.js:154:12)
    at Runner.query (/Users/user1/strapi4-demo/node_modules/knex/lib/execution/runner.js:141:36)
    at ensureConnectionCallback (/Users/user1/strapi4-demo/node_modules/knex/lib/execution/internal/ensure-connection-callback.js:13:17)
    at Runner.ensureConnection (/Users/user1/strapi4-demo/node_modules/knex/lib/execution/runner.js:318:20)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Runner.run (/Users/user1/strapi4-demo/node_modules/knex/lib/execution/runner.js:30:19)
    at async Object.execute (/Users/user1/strapi4-demo/node_modules/@strapi/database/lib/query/query-builder.js:481:22)
    at async Object.create (/Users/user1/strapi4-demo/node_modules/@strapi/database/lib/entity-manager/index.js:235:19)
```

**Comment by wusuopu** [comment]
maybe the `SqliteError` is due to `"useJoinTable": false` config. When I remove it, that is ok.
But I don't want to create the third table.

**Comment by gu-stav** [comment]
The `id` normalization was fixed in 4.12 via https://github.com/strapi/strapi/pull/17340.

**Comment by derrickmehaffy** [comment]
> maybe the `SqliteError` is due to `"useJoinTable": false` config. When I remove it, that is ok. But I don't want to create the third table.

for the most part joining tables are required, that config option was never designed to be used by users it was for some special cases internally.

Closing this as it's an unsupported config option
